client/src/components/survey/Survey.jsx

```diff
@@ -96,23 +99,49 @@ const useStyles = makeStyles(() => ({
 }));
 
 const Survey = props => {
-  const { surveyPage } = props;
+  const { surveyPage, setSurveyPage, requiredStep } = props;
   const classes = useStyles();
   const [activeStep, setActiveStep] = useState(surveyPage - 1);
+  const [open, setOpen] = useState(false);
```

Comment:
Let's rename these to something more meaningful: `openWarning`, `openWarningDialog`, `showWarningDialog`, something like that